xcdat/axis.py

```diff
@@ -0,0 +1,204 @@
+        if axis == "lat":
+            bounds = self._calc_lon_bounds(bounds)
+
+        # Compose final DataArray to set as a data variable
+        axis_bnds_var = xr.DataArray(
+            data=bounds,
+            coords={axis: axis_coords.data},
+            dims=[axis, "bnds"],
+            attrs={"units": axis_coords.units, "is_calculated": True},
+        )
+        self._obj[f"{axis}_bnds"] = axis_bnds_var
+        return axis_bnds_var
+
+    def _extract_axis_coords(self, axis: Axis) -> xr.DataArray:
+        """Extracts the coordinates for the specified axis.
+
+        :param axis: "lat" or "lon" axis
+        :type axis: Axis
+        :raises KeyError: If an incorrect "axis" argument is passed
+        :return: Matching coordinates
+        :rtype: xr.DataArray
+        """
+        axis_coord_names = AxisAccessor.axes_map[axis]["coords"]
+        matching_coords = None
+
+        for name in axis_coord_names:
+            matching_coords = self._obj.coords.get(name)
+            if matching_coords is not None:
+                break
+
+        if matching_coords is None:
+            raise TypeError(f"No matching coordinates for axis: {axis}")
+
+        return matching_coords
+
+    def _gen_base_bounds(
+        self,
+        axis_coords: xr.DataArray,
+        width: float,
+    ) -> np.ndarray:
+        """Generates the base 2D bounds array based on the axis coordinates variable.
+
+        :param axis_coords: The axis coordinates variable
+        :type axis_coords: xr.DataArray
+        :param width: Width of the bounds
+        :type width: float
+        :return: [description]
+        :rtype: np.ndarray
+        """
+        axis_coords_data = axis_coords.data
+
+        if len(axis_coords_data) > 1:
+            left = np.array([1.5 * axis_coords_data[0] - 0.5 * axis_coords_data[1]])
+            center = (axis_coords_data[0:-1] + axis_coords_data[1:]) / 2.0
+            right = np.array([1.5 * axis_coords_data[-1] - 0.5 * axis_coords_data[-2]])
+
+            bounds = np.concatenate((left, center, right))
+        else:
+            delta = width / 2.0
+            bounds = np.array([axis_coords[0] - delta, axis_coords[0] + delta])
+
+        bounds_2d = np.zeros((axis_coords_data.shape + (2,)), np.float64)
+        bounds_2d[..., 0] = bounds[:-1]
+        bounds_2d[..., 1] = bounds[1:]
+
+        return bounds_2d
+
+    def _calc_lat_bounds(self, bounds_2d: np.ndarray) -> np.ndarray:
+        """Calculate latitude boundaries and avoids floating point errors.
+
+        :param bounds_2d: [description]
+        :type bounds_2d: np.ndarray
+        :return: [description]
+        :rtype: np.ndarray
+
+        # TODO: If possible, refactor this method to simplify conditionals
+        """
+        min_neg_val = bounds_2d[0, 0]
+        min_pos_val = bounds_2d[0, 1]
+        second_max_pos_val = bounds_2d[-1, 0]
+        max_pos_val = bounds_2d[-1, 1]
+        max_degree_interval = 360.0
```

Comment:
Let me know if these variable names make sense. I just checked what the values are and named them accordingly.

Reply:
IMHO min_left/right and max_left/right might read better.

Reply:
Thanks for the suggestion, I agree and adopted this convention